File: src/git/worktrees/contracts.ts
export interface ManagedWorktree {
  id: string; name: string; rootPath: string; repositoryPath: string; commonDir: string;
  branch: string; baseBranch: string; baseHead: string; state: "creating" | "ready" | "incomplete" | "removed";
}
export interface ExistingWorktree { rootPath: string; branch: string | null; head: string; locked: boolean }
export interface WorktreeReport { rootPath: string; branch: string; branches: string[]; worktrees: ExistingWorktree[]; managed: ManagedWorktree[] }
export interface CreateInput { rootPath: string; name: string; branch: string; baseBranch: string }
export type WorktreeAction = "commit" | "merge" | "cleanup";
export interface ActionInput { id: string; action: WorktreeAction; message?: string; targetPath?: string }
export interface WorktreePreview { token: string; title: string; rootPath: string; branch: string; targetPath: string | null; files: string[]; diff: string; notes: string[] }
export interface ActionResult { message: string; worktree: ManagedWorktree | null; conflicts: string[] }

File: src-tauri/src/git/worktrees/contracts.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ManagedWorktree {
    pub id: String, pub name: String, pub root_path: String, pub repository_path: String,
    pub common_dir: String, pub branch: String, pub base_branch: String,
    pub base_head: String, pub state: String,
}
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ExistingWorktree {
    pub root_path: String, pub branch: Option<String>, pub head: String, pub locked: bool,
}
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct WorktreeReport {
    pub root_path: String, pub branch: String, pub branches: Vec<String>,
    pub worktrees: Vec<ExistingWorktree>, pub managed: Vec<ManagedWorktree>,
}
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateInput { pub root_path: String, pub name: String, pub branch: String, pub base_branch: String }

#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum Action { Commit, Merge, Cleanup }
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ActionInput {
    pub id: String, pub action: Action, pub message: Option<String>, pub target_path: Option<String>,
}
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct WorktreePreview {
    pub token: String, pub title: String, pub root_path: String, pub branch: String,
    pub target_path: Option<String>, pub files: Vec<String>, pub diff: String,
    pub notes: Vec<String>,
}
#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ActionResult { pub message: String, pub worktree: Option<ManagedWorktree>, pub conflicts: Vec<String> }
